src/engine/strat_engine/blockdev.rs

```diff
@@ -112,10 +123,57 @@ impl BlockDev {
     pub fn max_metadata_size(&self) -> Sectors {
         self.bda.max_data_size()
     }
+
+    /// Timestamp when the device was initialized.
+    pub fn initialization_time(&self) -> u64 {
+        self.bda.initialization_time()
+    }
+}
+
+impl HasUuid for StratBlockDev {
+    fn uuid(&self) -> &DevUuid {
+        &self.bda.dev_uuid()
+    }
+}
+
+impl BlockDev for StratBlockDev {
+    fn devnode(&self) -> PathBuf {
+        self.devnode.clone()
+    }
+
+    fn user_id(&self) -> &Option<String> {
+        &self.location
+    }
+
+    fn set_user_id(&mut self, location: Option<&str>) -> EngineResult<()> {
+        self.location = location.map(|x| x.to_owned());
+        Ok(())
+    }
+
+    fn hardware_id(&self) -> &Option<String> {
+        &self.disk_id
+    }
+
+    fn initialization_time(&self) -> String {
+        Utc.timestamp(self.initialization_time() as i64, 0)
+            .to_rfc3339()
+    }
+
+    fn total_size(&self) -> Sectors {
+        self.bda.dev_size()
+    }
+
+    fn state(&self) -> BlockDevState {
+        BlockDevState::InUse
```

Comment:
This can't be right. Doesn't it need a TODO or explanation of some sort?

Reply:
ok